File: fixtures/scale/dart/large/lib/ciphers/railfence/rail_pattern.dart
/// Maps character positions onto rails of the zigzag fence.
class RailPattern {
  final int rails;

  RailPattern(this.rails);

  int railCount() => rails;

  int railFor(int index) {
    final cycle = 2 * (rails - 1);
    final pos = index % cycle;
    return pos < rails ? pos : cycle - pos;
  }
}

File: fixtures/scale/kotlin/large/src/cryptobench/ciphers/railfence/RailPattern.kt
package cryptobench.ciphers.railfence

/** Maps character positions onto rails of the zigzag fence. */
internal class RailPattern(private val rails: Int) {

    fun railCount(): Int = rails

    fun railFor(index: Int): Int {
        val cycle = 2 * (rails - 1)
        val pos = index % cycle
        return if (pos < rails) pos else cycle - pos
    }
}
